pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -603,47 +603,37 @@ func flavorSelector(spec *corev1.PodSpec, allowedKeys sets.Set[string]) nodeaffi
 func (a *FlavorAssigner) fitsResourceQuota(log logr.Logger, fr resources.FlavorResource, val int64, rQuota cache.ResourceQuota) (granularMode, bool, *Status) {
 	var status Status
 	var borrow bool
-	used := a.cq.ResourceNode.Usage[fr]
-	mode := noFit
-	if val <= rQuota.Nominal {
-		// The request can be satisfied by the nominal quota, assuming quota is
-		// reclaimed from the cohort or assuming all active workloads in the
-		// ClusterQueue are preempted.
-		mode = preempt
-	}
 
-	if a.canPreemptWhileBorrowing() {
-		// when preemption with borrowing is enabled, we can succeed to admit the
-		// workload if preemption is used.
-		if (rQuota.BorrowingLimit == nil || val <= rQuota.Nominal+*rQuota.BorrowingLimit) && val <= a.cq.PotentialAvailable(fr) {
-			mode = preempt
-			borrow = val > rQuota.Nominal
-		}
-	}
-	if rQuota.BorrowingLimit != nil && used+val > rQuota.Nominal+*rQuota.BorrowingLimit {
-		status.append(fmt.Sprintf("borrowing limit for %s in flavor %s exceeded", fr.Resource, fr.Flavor))
-		return mode, borrow, &status
-	}
+	available := a.cq.Available(fr)
+	maxCapacity := a.cq.PotentialAvailable(fr)
 
-	if a.oracle.IsReclaimPossible(log, a.cq, *a.wl, fr, val) {
-		mode = reclaim
+	// No Fit
+	if val > maxCapacity {
+		status.append(fmt.Sprintf("insufficient quota for %s in flavor %s, request > maximum capacity (%s > %s)",
+			fr.Resource, fr.Flavor, resources.ResourceQuantityString(fr.Resource, val), resources.ResourceQuantityString(fr.Resource, maxCapacity)))
+		return noFit, false, &status
 	}
 
-	lack := val - a.cq.Available(fr)
-	if lack <= 0 {
-		return fit, used+val > rQuota.Nominal, nil
+	// Fit
+	if val <= available {
+		return fit, a.cq.ResourceNode.Usage[fr]+val > rQuota.Nominal, nil
 	}
 
-	lackQuantity := resources.ResourceQuantity(fr.Resource, lack)
-	msg := fmt.Sprintf("insufficient unused quota in cohort for %s in flavor %s, %s more needed", fr.Resource, fr.Flavor, &lackQuantity)
-	if !a.cq.HasParent() {
-		if mode == noFit {
-			msg = fmt.Sprintf("insufficient quota for %s in flavor %s in ClusterQueue", fr.Resource, fr.Flavor)
-		} else {
-			msg = fmt.Sprintf("insufficient unused quota for %s in flavor %s, %s more needed", fr.Resource, fr.Flavor, &lackQuantity)
+	// Check if preemption is possible
+	mode := noFit
+	if val <= rQuota.Nominal {
+		mode = preempt
+		if a.oracle.IsReclaimPossible(log, a.cq, *a.wl, fr, val) {
+			mode = reclaim
 		}
+	} else if a.canPreemptWhileBorrowing() {
+		mode = preempt
+		borrow = true
 	}
-	status.append(msg)
+
+	status.append(fmt.Sprintf("insufficient unused quota for %s in flavor %s, %s more needed",
```

Comment:
Do we want to set this message even if `val <= rQuota.Nominal`?

Reply:
We did this (a similar message) before, and I think the message is still relevant/accurate - other quota is being used